otter/convergence/transforming.py

```diff
@@ -91,6 +92,30 @@ def optimize_steps(rcv3_steps):
 _register_bulk_rcv3_optimizer(BulkRemoveFromRCv3)
 
 
+def filter_clb_mutating_types(steps):
+    """
+    Allow only one CLB mutating steps per CLB
+
+    :param steps: Iterable of :obj:`IStep` instances
+    :return: Iterable of :obj:`IStep` instances such that only one mutating
+        type is returned per CLB
+    """
+    mutating_clb_types = (AddNodesToCLB, RemoveNodesFromCLB, ChangeCLBNode)
+    lb_step_type = {}
+    for step in steps:
+        stype = type(step)
+        if stype not in mutating_clb_types:
+            yield step
+            continue
+        lb_stype = lb_step_type.get(step.lb_id)
+        if lb_stype is None:
+            lb_step_type[step.lb_id] = stype
```

Comment:
This algorithm will allow multiple concurrent modifications _of the same type_, which I think is not what's intended.  As you explained to me on slack; `AddNodesToCLB` and `RemoveNodesFromCLB` will be coalesced into single batch steps, but `ChangeCLBNode` won't (can't) be, which means this could still allow through multiple concurrent `ChangeCLBNode` calls, and that would leave us in the same problematic scenario we now find ourselves.

The good news is that the algorithm actually gets quite a bit simpler if you implement it to simply base it on LB ID rather than LB ID plus type; something like this should work:

```python
def filter_clb_mutating_types(steps):
    lbs_with_mutations = set()  # set of loadbalancer IDs
    for step in steps:
        if isinstance(step, (AddNodesToCLB, RemoveNodesFromCLB,
                             ChangeCLBNode)):
            if step.lb_id in lbs_with_mutations:
                continue
            lbs_with_mutations.add(step.lb_id)
        yield step
```